packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -88,19 +88,19 @@ export function callTracedServerSideDataFetcher<F extends (...args: any[]) => Pr
 
     if (requestTransaction === undefined) {
       const sentryTraceHeader = req.headers['sentry-trace'];
-      const rawBaggageString = req.headers && isString(req.headers.baggage) && req.headers.baggage;
+      const rawBaggageString = req.headers && req.headers.baggage;
       const traceparentData =
         typeof sentryTraceHeader === 'string' ? extractTraceparentData(sentryTraceHeader) : undefined;
 
-      const baggage = parseBaggageSetMutability(rawBaggageString, traceparentData);
+      const dynamicSamplingContext = baggageHeaderToDynamicSamplingContext(rawBaggageString);
 
       const newTransaction = startTransaction({
         op: 'nextjs.data.server',
         name: options.requestedRouteName,
         ...traceparentData,
         metadata: {
           source: 'route',
-          baggage,
+          dynamicSamplingContext: traceparentData && !dynamicSamplingContext ? {} : dynamicSamplingContext,
```

Comment:
if we have an invalid `traceparentData`, do we want to even include a `dynamicSamplingContext`? Setting the `dynamicSamplingContext` to be an empty object will freeze this, which doesn't make a ton of sense to me.

We could either:
a) not freeze baggage if it comes in as an empty object (invalid - didn't have a `trace_id` or `public_key`)
b) make sure that we only define `dynamicSamplingContext` if traceparentData is defined. This means we still have the possibility of an empty object, but should be fine since that means the parent sdk just messed something up.

I'm leaning toward b - so I would do something like so:

```ts
...(traceparentData && dynamicSamplingContext && { dynamicSamplingContext }),
```

We could also construct a helper for this, since we repeat this logic in some other places.

Reply:
The reason for this ternary here was to freeze baggage if there is a `sentry-trace` header, but no `baggage` header. The freezing is intentional and to handle requests of old SDKs correctly. I confirmed with Jan that we want to keep this behavior. Sadly, this means we can't do the thing you suggested.

In normal circumstances, it shouldn't happen that there is a `baggage` header with `sentry-` values but no `sentry-trace` header. This is why I didn't put any handling for that case here.

Do you have any other thoughts regarding this, otherwise I would leave it as is. Having a helper seems overkill to me for a simple if statement (even though I admit it is repeated in a lot of places).

What I can do, is add a comment where we do this to explain that DSC needs to be frozen when there is trace parent data. Wdyt?

Reply:
> The reason for this ternary here was to freeze baggage if there is a sentry-trace header, but no baggage header

Ahhhh I see - this makes sense. Let me try playing around with the helper to clean it up a little.

Reply:
Ok I give up - I think we keep this for now then.